src/domains/search/components/SearchResult.tsx

```diff
@@ -138,11 +140,17 @@ export const SearchResult = forwardRef(function SearchResult(
         });
       }
 
-      if (urlObject?.pathname) {
+      if (
+        urlObject?.pathname &&
+        (feature.osmFeature !== null || feature.placeInfo !== null)
+      ) {
         push(urlObject);
+      } else {
+        replace({ pathname: "/" });
+        resetSearchTerm("");
```

Comment:
The `SearchResult` should not be able to set the search term. The responsibility of the search query lies in the parent `Search` component. In this case I would rather add a `onClick` event listener prop to the `SearchResult` and handle resetting the search query in the parent component. 🙂